tsdb/head_test.go

```diff
@@ -4792,9 +4792,11 @@ func TestWBLReplay(t *testing.T) {
 	require.False(t, ok)
 	require.NotNil(t, ms)
 
-	xor, err := ms.ooo.oooHeadChunk.chunk.ToXOR()
+	chks, err := ms.ooo.oooHeadChunk.chunk.ToEncodedChunks(math.MinInt64, math.MaxInt64)
```

Comment:
TestWBLReplay could be refactored to use scenarios. I assume that's coming in followup PR? ( i did a little test and enabled scenarios and could get TestOutOfOrderSamplesMetric to pass with an additional `db.EnableNativeHistograms()` statement)

Note to self for that review: let's detach the value of the sample from its timestamp to be able to insert a lower value after a higher one to trigger counter reset detection.

So instead of 
```
appendSample(50, true)
```
you'd have something like

```
appendSample(50, 20, true)
```

Reply:
I went ahead and added that to this PR :) The introduction of counter resets can come alongside the rest of the OOO native histogram support